xilinx/common/tcl/prologue.tcl

```diff
@@ -116,3 +119,7 @@ if {[get_filesets -quiet constrs_1] eq ""} {
 
 set obj [current_fileset -constrset]
 add_files -norecurse -fileset $obj [glob -directory $constraintsdir {*.xdc}]
+
+if {[info exists addl_constraint_dir]} {
+    add_files -norecurse -fileset $obj [glob -directory $addl_constraint_dir {*.xdc}]
+}
```

Comment:
Maybe to DRY this up, we should replace `$constraintsdir` with `$constraintsdirs` (plural), make it a list, and then conditionally add the `addl-constraint-dir` if it is provided? Then these lines would be to loop over the list of `$constraintsdirs` and add the files in.